pkg/sip/server.go

```diff
@@ -132,8 +139,10 @@ type Server struct {
 	sipListeners []io.Closer
 	sipUnhandled RequestHandler
 
-	imu               sync.Mutex
-	inProgressInvites []*inProgressInvite
+	imu                  sync.Mutex
+	inProgressInvites    map[dialogKey]*inProgressInvite
+	inviteTimeoutQueue   utils.TimeoutQueue[*dialogKey]
```

Comment:
```suggestion
	inviteTimeoutQueue   utils.TimeoutQueue[dialogKey]
```
This must be without a pointer. Otherwise you are using the pointer address as a queue key, so it will never match.